formstack-baton-requests/src/main/scala/com/gu/identity/formstackbatonrequests/FormstackSarHandler.scala

```diff
@@ -0,0 +1,47 @@
+package com.gu.identity.formstackbatonrequests
+
+import java.util.UUID.randomUUID
+
+import com.gu.identity.formstackbatonrequests.BatonModels.{Completed, Failed, Pending, SarInitiateRequest, SarInitiateResponse, SarPerformRequest, SarRequest, SarResponse, SarStatusRequest, SarStatusResponse}
+import com.gu.identity.formstackbatonrequests.lambda.LambdaClient
+import com.gu.identity.formstackbatonrequests.s3.S3Client
+import com.typesafe.scalalogging.LazyLogging
+
+case class FormstackSarHandler(s3Client: S3Client, lambdaClient: LambdaClient, sarHandlerConfig: SarLambdaConfig)
+  extends LazyLogging
+    with FormstackHandler[SarRequest, SarResponse] {
+
+  private def initiate(request: SarInitiateRequest): Either[Throwable, SarInitiateResponse] = {
+    val initiationReference = randomUUID.toString
+
+    val performSarRequest = SarPerformRequest(
+      initiationReference,
+      request.subjectEmail
+    )
+
+    logger.info(s"invoking FormstackPerformSarLambda with initiation reference: $initiationReference")
+    lambdaClient.invokeLambda(performSarRequest, sarHandlerConfig)
```

Comment:
why does this lambda call another lambda?

Reply:
This is how all the lambda's for baton have been working (excluding the Baton lambda which can use a query ID returned by Athena). 1st lambda only deals with status checking (in S3) and initiating the lambda that actually performs with SAR. Then the second one performs the actual SAR and writes to S3.